### PR TITLE
boards: Remove PINCTRL from some boards

### PR DESCRIPTION
**Pull request by nordicjm**
Removes lines from defconfig files having PINCTRL in them

### CODE REVIEW

File: boards/arduino/nicla_sense_me/arduino_nicla_sense_me_defconfig [hunk 1]
```diff
@@ -13,6 +13,3 @@ CONFIG_SERIAL=y
 # enable console
 CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
-
-# enable pin controller
-CONFIG_PINCTRL=y
```
Comment (jonas-rem) [inline boards/arduino/nicla_sense_me/arduino_nicla_sense_me_defconfig:18]: Same here, this is a board based on a Nordic SoC.

**Review by jonas-rem — CHANGES_REQUESTED** [review]

File: boards/phytec/phyboard_electra/phyboard_electra_am6442_m4_defconfig [hunk 2]
```diff
@@ -11,9 +11,6 @@ CONFIG_CORTEX_M_SYSTICK=y
 # Zephyr Kernel Configuration
 CONFIG_XIP=n
 
-# Enable Pinctrl
-CONFIG_PINCTRL=y
-
```
[the remark this answers is not in the document]
Comment (nordicjm) [reply boards/phytec/phyboard_electra/phyboard_electra_am6442_m4_defconfig:16]: Renamed

**Review by jonas-rem — APPROVED** [review]

**Review by dnltz — APPROVED** [review]
Thanks for removing them!

**Comment by dnltz** [comment]
@nordicjm I don't have the permissions to merge this pull request. Where can I find out who I should assign this PR to?

**Comment by nordicjm** [comment]
It will be merged post zephyr 4.0 release by someone with merge rights (currently zephyr is in feature free for 4.0)

**Comment by dnltz** [comment]
> It will be merged post zephyr 4.0 release by someone with merge rights (currently zephyr is in feature free for 4.0)

@nordicjm thank you for the explaining!

**Pull request merged**